schema/elements.go

```diff
@@ -110,6 +107,33 @@ type Struct struct {
 	ElementRelationship ElementRelationship `yaml:"elementRelationship,omitempty"`
 }
 
+// Union, or oneof, means that only one of multiple fields of a structure can be
+// set at a time. For backward compatibility reasons, and to help "dumb clients"
+// which are not aware of the union (or can't be aware of it because they
+// don't know what fields are part of the union), the code tolerates multiple
+// fields to be set but will try to detect which fields must be cleared (there
+// should never be more than two though):
+// - If there is a discriminator and its value has changed, clear all fields
+// but the one specified by the discriminator
+// - If there is no discriminator, or it hasn't changed, if new has two of the
+// fields set, remove the one that was set in old.
+// - If there is a discriminator, set it to the value we've kept (if it changed)
+type Union struct {
+	// Discriminator, if present, is the name of the field that
+	// discriminates fields in the union. The mapping between the value of
+	// the discriminator and fields works as by uncapitalizing the first
+	// letter (or leave untouched if not capitalized).
+	Discriminator *string `yaml:"discriminator,omitempty"`
+
+	// This is the list of fields that belong to this union. This fields are
+	// required to not be part of another union, or be the discriminator for
+	// another union. All the fields present in here have to be part of the
+	// parent structure. Discriminator (if oneOf has one), is NOT included
+	// in this list. The value for field is how we map the name of the field
+	// to actual value for discriminator.
+	Fields map[string]string `yaml:"fields,omitempty"`
```

Comment:
can a value be mapped to from multiple keys?

Reply:
Also maybe this should be a list, since we use lists in all the other schema pieces

Reply:
> Also maybe this should be a list, since we use lists in all the other schema pieces

Good point

> can a value be mapped to from multiple keys?

No, it shouldn't. The part that converts the openapi into this schema is going to verify that this is not possible. If it still happens that we have to mappings going to the same key, it will probably use the last one in the list.